### PR TITLE
fix: Modify `__ERROR` to accept string literal

### PR DESCRIPTION
**Pull request by Sabnock01**
## Overview

This PR modifies `__ERROR` to accept a string literal as an argument the same as `__FUNC_SIG`.

### CODE REVIEW

**Comment by Sabnock01** [comment]
Actually after reading over [this Solidity blog post](https://blog.soliditylang.org/2021/04/21/custom-errors/) current error padding seems correct so probably that shouldn't be changed. Or maybe we'd like the function selector to be padded the same as errors instead of the other way around. Not sure. What do we think? 

Am fine with abandoning the padding in favor of only adding the string literal to `__ERROR`

**Comment by Sabnock01** [comment]
Went ahead and removed. Would prefer minimal review on this one.

**Review by clabby — DISMISSED** [review]
Nice!

File: huff_codegen/src/irgen/statements.rs [hunk 1]
```diff
@@ -431,6 +431,18 @@ pub fn statement_gen(
...
+                    } else if let Some(s) = &bf.args[0].name {
+                        let mut signature = [0u8; 4]; // Only keep first 4 bytes
+                        hash_bytes(&mut signature, s);
+
+                        let push_bytes = format!(
+                            "{}{}{}",
+                            Opcode::Push32,
+                            "00".repeat(28),
+                            hex::encode(signature)
+                        );
```
Comment (clabby) [inline huff_codegen/src/irgen/statements.rs:443]:
```suggestion
                        let push_bytes = format!(
                            "{}{}",
                            Opcode::Push4,
                            hex::encode(signature)
                        );
```

File: huff_codegen/src/irgen/statements.rs [hunk 2]
```diff
@@ -431,6 +431,18 @@ pub fn statement_gen(
...
                         *offset += push_bytes.len() / 2;
                         bytes.push((starting_offset, Bytes(push_bytes)));
+                    } else if let Some(s) = &bf.args[0].name {
+                        let mut signature = [0u8; 4]; // Only keep first 4 bytes
+                        hash_bytes(&mut signature, s);
+
+                        let push_bytes = format!(
+                            "{}{}{}",
+                            Opcode::Push32,
+                            "00".repeat(28),
```
[the remark this answers is not in the document]
Comment (Sabnock01) [reply huff_codegen/src/irgen/statements.rs]: Good call

**Review by clabby — APPROVED** [review]

**Pull request merged**